src/mpi/coll/transports/mpich/api_def.h

```diff
@@ -0,0 +1,50 @@
+#define TRANSPORT_NAME MPICH_
```

Comment:
It is much more aligned with way netmods/devices do that.  
Collectives templates uses TSP_ api , and this API contains of interface functions only. Transports' internals should not be exposed via TSP_ interface.
Additionally, that makes code much more readable and easy to manage.